compiler/types.nim

```diff
@@ -1500,7 +1503,7 @@ proc compatibleEffects*(formal, actual: PType): EffectsCompat =
 
 
 proc isCompileTimeOnly*(t: PType): bool {.inline.} =
-  result = t.kind in {tyTypeDesc, tyStatic}
+  result = t.kind in {tyTypeDesc, tyStatic, tyGenericParam}
```

Comment:
This change exists in this PR for the sole purpose of making `T; U: static T` work (though might be more correct anyway). In any case the compiler treats `static T` the same as `static foo()` and evaluates it as a constant expression which returns `static[T]`. `static[T]` on the other hand goes through `semStaticType` in `semtypes` which doesn't have a reason to fail with generic params. I didn't make a more sweeping change in line with this to lessen the scope of this PR.